src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -40,11 +40,43 @@ struct InternalFlashFactoryData
         return CHIP_NO_ERROR;
     }
 
+#ifdef CONFIG_CHIP_FACTORY_DATA_WRITE_PROTECT
+#define TO_STR_IMPL(x) #x
+#define TO_STR(x) TO_STR_IMPL(x)
+    // These two helpers allows to get the factory data memory block which shall be protected with fprotect, so that:
+    // 1) it is aligned to the multiple of CONFIG_FPROTECT_BLOCK_SIZE (which differs between nRF families)
+    // 2) it does not exceed the settings partition start address
+    // Note that this block can overlap with app partition but this is not a problem since we do not aim to modify
+    // the application code at runtime anyway.
+    constexpr size_t FactoryDataBlockBegin()
+    {
+        // calculate the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE smaller than PM_FACTORY_DATA_ADDRESS
+        return PM_FACTORY_DATA_ADDRESS & (-CONFIG_FPROTECT_BLOCK_SIZE);
+    }
+
+    constexpr size_t FactoryDataBlockSize()
+    {
+        // calculate the address which includes the PM_FACTORY_DATA_SIZE but give a chance to add another offset
```

Comment:
"calculate the factory data end address rounded up to the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE" would be a bit more precise.